File: ISA_BACKEND/BloodBank/src/main/java/com/example/bloodbank/Model/CenterVisit.java
package com.example.bloodbank.Model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "CenterVisit")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class CenterVisit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private long id;
    @ManyToOne
    @JoinColumn(name = "blood_donor_id")
    private BloodDonor bloodDonor;
    @OneToOne
    @JoinColumn(name = "blood_donation_id")
    private BloodDonationAppointment bloodDonationAppointment;
    private float price;
    private boolean isCanceled;
}

File: ISA_FRONTEND/src/app/model/centerVisit.ts
import { BloodDonationAppointment } from "./bloodDonationAppointment";
import { BloodDonor } from "./bloodDonor";

export class CenterVisit {
    id: string = '';
    bloodDonationAppointment: BloodDonationAppointment = new BloodDonationAppointment;
    bloodDonor : BloodDonor = new BloodDonor;
    price : number = 0;

    public constructor(obj?: any) {
        if (obj) {
            this.id = obj.id;
            this.price = obj.price;
            this.bloodDonationAppointment = obj.bloodDonationAppointment;
            this.bloodDonor = obj.bloodDonor;
        }
    }
}